src/config.rs

```diff
@@ -319,6 +321,19 @@ fn construct_proxy_config(mc_path: &str, pc_env: Option<&str>) -> anyhow::Result
         .filter_map(|(k, v)| Some((k.strip_prefix("")?.to_string(), v)))
         .collect();
 
+    let istio_env_vars: Vec<(String, String)> = env::vars()
+        .filter(|(key, _)| key.starts_with(ISTIO_META_PREFIX))
+        .map(|(key, val)| {
+            (
+                key.trim_start_matches(ISTIO_META_PREFIX).to_lowercase(),
```

Comment:
Why lowercase?

Also I think L321 needs the stripping as well (see envoy logic: https://github.com/istio/istio/blob/b5df9a51c0290df8fd2692e7578bbef62a315d42/pkg/bootstrap/config.go#L563)

Reply:
Fixed. It should not be lowercased. Just added these code logic https://github.com/istio/istio/blob/b5df9a51c0290df8fd2692e7578bbef62a315d42/pkg/bootstrap/config.go#L567-L576. However, the metadata struct defined here is `HashMap<String, String>`, which is not equal to `map[string]any`, should I revise it and add more logics or leave it like this and do it later when needed?